usr/libexec/greenboot/greenboot.sh

```diff
@@ -2,75 +2,56 @@
 set -euo pipefail
 IFS=$'\n\t'
 
-run_required_scripts () {
-  echo "Running Required Health Check Scripts..."
-  local required_scripts=`find /etc/greenboot/check/required.d -name '*.sh'`
+script_runner () {
+  if [ $# -eq 0 ]; then
+    echo "Please call '$0 run <dir_path> [{strict,relaxed}] [message]' to run this command!"
+    exit 1
+  fi
   local rc=0
-  for script in $required_scripts; do
+  local scripts=`find $1 -name '*.sh'`
+  case "${2:-s}" in
+    "r" | "relaxed")
+      local mode=relaxed
+      ;;
+    "s" | "strict" | *)
+      local mode=strict
+      ;;
+  esac
+  local start_msg=${3:-Running scripts from $1 in $mode mode...}
+  echo "$start_msg"
+  for script in $scripts; do
```

Comment:
You'll want to make sure you always reset `rc` here before running the script. Otherwise, the first fail will mark all following scripts as fails too!

Reply:
whoops!